cs50/2017/fall/recover/check50/__init__.py

```diff
@@ -74,23 +128,43 @@ def first_image(self):
         """recovers 000.jpg correctly"""
         self.add("card.raw")
         self.spawn("./recover card.raw").exit(0, timeout=10)
-        if self.hash("000.jpg") != self.hashes[0]:
-            raise Error("recovered image does not match")
+        self.check_jpg(0)
 
     @check("compiles")
     def middle_images(self):
         """recovers middle images correctly"""
         self.add("card.raw")
         self.spawn("./recover card.raw").exit(0, timeout=10)
-
-        for i, hash in enumerate(self.hashes[1:-1], 1):
-            if hash != self.hash("{:03d}.jpg".format(i)):
-                raise Error("recovered image does not match")
+        for i in range(49):
+            self.check_jpg(i)
 
     @check("compiles")
     def last_image(self):
         """recovers 049.jpg correctly"""
         self.add("card.raw")
         self.spawn("./recover card.raw").exit(0, timeout=10)
-        if self.hash("049.jpg") != self.hashes[-1]:
-            raise Error("recovered image does not match")
+        self.check_jpg(49)
+
+    def check_jpg(self, i):
```

Comment:
Do add comments throughout this method, including a docstring under the `def` so that future staff can easily understand what it's doing and how.